src/decoder/stream.rs

```diff
@@ -758,25 +741,36 @@ impl StreamingDecoder {
                         type_: _,
                     } = &mut self.current_chunk;
 
+                    if raw_bytes.len() == raw_bytes.capacity() {
+                        if self.limits.bytes == 0 {
+                            return Err(DecodingError::LimitsExceeded);
+                        }
+
+                        // Double the size of the Vec, but not beyond the allocation limit.
+                        debug_assert!(raw_bytes.capacity() > 0);
+                        let reserve_size = raw_bytes.capacity().min(self.limits.bytes);
```

Comment:
I think the total size of the chunk should also be available at this point. You could include that in the calculation, to avoid the doubling overshooting the chunk size.